### DISCUSSION TITLE
Is it possible to load a plugin on page events?

### DISCUSSION THREAD

**Discussion by schulle4u** [datenstrom/community#261] [Ask a question] [answered]
If this is actually possible, what would be the best way to load a plugin only on certain Yellow events? The only solutions I could find in the API documentation are `$yellow->page->isActive()` to do something if a certain page is requested, and `$yellow->plugins->isExisting($name)` to do something if a certain plugin exists. In addition I'm looking for the possibility to check if a certain string is present within a page, for example a shortcut. 

To explain further: I'm currently working on a [Akismet plugin](https://github.com/schulle4u/yellow-plugin-akismet) to integrate it as a service for other plugins. My solution already works, however the plugin and it's included external class seems to slow down the page load. So my idea is to only load the plugin if for example the contact plugin is active on a page. Simply loading Akismet if the contact plugin is registered still wohld call it on all other pages too. 

Any ideas on this? Maybe I'm completely wrong with my current solution. 

Steffen

**Comment by markseuffert** [comment, marked as answer]
How about the event `onRequest()` from the API?

There you could check if with `$_REQUEST["status"]=="send"` if you have something to do and then change the status with` $_REQUEST["status"]="error"` to prevent further processing. This should in theory work with contact/comments plugin, but I haven't tested it.

Hope this gives you some starting points.

**Comment by schulle4u** [comment]
Thanks Mark, sounds promising. I'll have a look into it. 

Especially for the event functions it would be great to have a bit more example code or use cases in the API documentation. Sometimes it is a little difficult to guess what some of the function parameters do. 

Steffen

**Comment by schulle4u** [comment]
I'm closing this for now, my question has been answered I think. Will do further work on Akismet when the comments plugin [has been fixed](https://github.com/nasendackel/yellow-comments/issues/16).

**Comment by markseuffert** [comment]
Just a quick followup. 

I added more information to the API documentation, especially for events. Now there's a sequence diagram, a description of a typical use case and three examples. In the beginning I considered more use cases, but it was hard to get an overview from a wall of text, so I opted for a sequence diagram instead. Hope this makes the learning curve easier.

Let me know what's missing. Feedback is always welcome.

**Comment by schulle4u** [comment]
I have to use a screen reader (braille and speech output) to access information, diagrams therefore are rather difficult to understand. Your descriptive text however is evident enough for me, so thanks for adding this to the docs. 

Steffen

**Comment by markseuffert** [comment]
I will send you an email with the other use cases. I kept a copy just in case. 😄 

Mark
